src/parser/SparqlParserHelpers.cpp

```diff
@@ -21,90 +21,55 @@ struct ParserAndVisitor {
  public:
   SparqlAutomaticParser _parser{&_tokens};
   SparqlQleverVisitor _visitor;
-  explicit ParserAndVisitor(string input) : _input{std::move(input)} {}
-  explicit ParserAndVisitor(string input,
-                            SparqlQleverVisitor::PrefixMap prefixMap)
-      : _input{std::move(input)}, _visitor{std::move(prefixMap)} {}
+  explicit ParserAndVisitor(string input) : _input{std::move(input)} {
+    _parser.setErrorHandler(std::make_shared<ThrowingErrorStrategy>());
+  }
 
   template <typename ResultType, typename ContextType>
-  auto parse(const std::string& input,
+  auto parse(const std::string& input, const std::string_view& name,
```

Comment:
`std::string_view` should be passed by value (it is a cheap const handle),
and probably both parameters should become `std::string_view`
(you then probably have to construct a `std::string` from the `remainingString`
for the constructor of `ResultOfParseAndRemainingText`.